File: Pie02/Pie02/Expressions/QualifiedImport.vb
Public Class QualifiedImport
    Inherits Expression

    Public Identifiers As New List(Of String)

    Public Sub New(parentExpression As Expression)
        MyBase.New(parentExpression)


    End Sub

    Public Shared Sub Build(parser As IronyParser, parentExpression As Expression, node As Irony.Parsing.ParseTreeNode)

        Dim import As New QualifiedImport(parentExpression)
        parentExpression.ChildExpressions.Add(import)

        For Each child In node.ChildNodes
            import.Identifiers.Add(child.FindTokenAndGetText())
        Next
    End Sub



    Public Overrides Sub Generate(ByRef source As String)
        source += "using "

        source += Identifiers(0)

        For i = 1 To Identifiers.Count - 1
            source += "." + Identifiers(i)
        Next

        source += ";" + Environment.NewLine
    End Sub

End Class

File: Pie02/Pie02IronPython/Expressions/QualifiedImport.py
from Expressions.Expression import *

class QualifiedImport(Expression):
    """description of class"""

    def __init__(self, parentExpression):
        super(self.__class__, self).__init__(parentExpression)
        self.Identifiers = []

    @staticmethod
    def Build(parser, parent, node):
        i = QualifiedImport(parent)
        parent.Children.append(i)
        for childNode in node.ChildNodes[0].ChildNodes:
            i.Identifiers.append(childNode.FindTokenAndGetText())

    def Emit(self, source):
        source += "using "
        source += self.Identifiers[0]
        for i in range(1, len(self.Identifiers)):
            source += '.'
            source += self.Identifiers[i]   
        source += ';\n'
        return source